Zend/zend_operators.c

```diff
@@ -1253,18 +1253,14 @@ ZEND_API zend_result ZEND_FASTCALL pow_function(zval *result, zval *op1, zval *o
 }
 /* }}} */
 
-#ifdef __clang__
-__attribute__((no_sanitize("float-divide-by-zero")))
-#endif
 static zend_result ZEND_FASTCALL div_function_base(zval *result, zval *op1, zval *op2) /* {{{ */
 {
 	zend_uchar type_pair = TYPE_PAIR(Z_TYPE_P(op1), Z_TYPE_P(op2));
 
 	if (EXPECTED(type_pair == TYPE_PAIR(IS_LONG, IS_LONG))) {
 		if (Z_LVAL_P(op2) == 0) {
-			zend_error(E_WARNING, "Division by zero");
-			ZVAL_DOUBLE(result, ((double) Z_LVAL_P(op1) / (double) Z_LVAL_P(op2)));
-			return SUCCESS;
+			zend_throw_error(zend_ce_division_by_zero_error, "Division by zero");
+			return FAILURE;
```

Comment:
I think these should all be SUCCESS returns. Only FAILURE if you can't handle the type. Otherwise we'll go through type conversion and try again.

Reply:
You probably also need `if (result != op1) { ZVAL_UNDEF(result); }

Reply:
> 
> 
> You probably also need `if (result != op1) { ZVAL_UNDEF(result); }`

Isn't that handled in ``div_function()`` ?

Reply:
div_function handles this or the case where it throws an exception. The same needs to happen in any place that throws. Btw, this is very likely the reason for the Zend/tests/bug76667.phpt failure.

You'll probably want to have a `div_by_zero:` once and then goto in other branches.